zebra-chain/src/block/commitment.rs

```diff
@@ -115,7 +115,7 @@ impl Commitment {
                 Ok(root) => Ok(FinalSaplingRoot(root)),
                 _ => Err(InvalidSapingRootBytes),
             },
-            Heartwood if Some(height) == Heartwood.activation_height(network) => {
+            Heartwood | Nu5 if Some(height) == Heartwood.activation_height(network) => {
```

Comment:
I think we always want to check if a height is the Heartwood activation height, but until now, we've known that it's only possible if the current network upgrade is Heartwood.

With the hard-coded activation heights we were using before, there were no cases where multiple network upgrades were activated at the same height, but currently on Regtest, `Genesis` is activated at height 0 and every other network upgrade is activated at height 1.

`NetworkUpgrade::current()` returns the latest network upgrade that's activated at the provided height, so currently on Regtest for heights above height 0, it returns is NU5. So it's now possible for a the current network upgrade to be NU5 (or Canopy, or any network upgrade above Heartwood) at the Heartwood activation height.

We may want to check if it's the Heartwood activation height for Canopy as well, I only added NU5 because the check is inefficient (though it may not have a measurable impact on performance).

Reply:
This is a great description of the general situation. I think you should include it somewhere in the codebase.

Reply:
Added in https://github.com/ZcashFoundation/zebra/commit/2ad072a19f1713090bbbaccabb5d3a7d381a55f1.